### PR TITLE
Fix dead assignment detected by clang CI

### PR DESCRIPTION
**Pull request by ghost**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

...

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

File: librz/io/p/io_rzpipe.c [hunk 1]
```diff
@@ -39,13 +38,8 @@ static int __write(RzIO *io, RzIODesc *fd, const ut8 *buf, int count) {
 		eprintf("rzpipe_write: error\n");
 		return -1;
 	}
-	res = rzpipe_read(RZP(fd));
```
Comment (ret2libc) [inline librz/io/p/io_rzpipe.c:42]: sure about this?? Maybe `res` is dead, but the call could have other effects, right?

File: librz/util/buf.c [hunk 2]
```diff
@@ -541,7 +541,6 @@ static st64 buf_format(RzBuffer *dst, RzBuffer *src, const char *fmt, int n) {
 			}
 
 			m = 1;
-			bigendian = true;
```
Comment (ret2libc) [inline librz/util/buf.c:544]: is this really dead? I'm not sure 100%

**Review by ret2libc — CHANGES_REQUESTED** [review]

File: librz/util/buf.c (hunk 2 above)
Comment (ghost) [reply librz/util/buf.c:544]: bigendian is assign in the switch line 461

Comment (ret2libc) [reply librz/util/buf.c:544]: it is not always set. Some cases do not set it and it may be used after that AFAIS.

Comment (ghost) [reply librz/util/buf.c:544]: case '0' - '9' -> continue (loop)
case 's', 'S', 'i', 'I', 'l', 'L', 'c' -> set bigendian
default -> return from the function

Comment (ghost) [reply librz/util/buf.c:544]: There is no side effect if the line 544 is removed

Comment (ret2libc) [reply librz/util/buf.c:544]: What happens if there is a number and then the fmt string ends? In that case, the previously set bigendian value would be considered, no?

Comment (ghost) [reply librz/util/buf.c:544]: bigendian is a local loop variable (line 458)

So the case of a number and then fmt strings ends as no impact

Comment (ghost) [reply librz/util/buf.c:544]: I don't see what I am missing.

Comment (ret2libc) [reply librz/util/buf.c:544]: Ok I think you are right!

**Review by ret2libc — APPROVED** [review]

**Pull request merged**